client/my-sites/plans-features-main/hooks/use-plan-type-destination-callback.ts

```diff
@@ -0,0 +1,58 @@
+import { plansLink } from '@automattic/calypso-products';
+import { useCallback } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+// TODO: Consider moving type reference
+import { PlanTypeSelectorProps } from 'calypso/my-sites/plans-grid/components/plan-type-selector/types';
+
+interface PathArgs {
+	[ key: string ]: string | null;
+}
+
+/**
+ * Returns a callback to retrieve a plan type destination path or query param depending on
+ * the environment ( Ex. ?intervalType=yearly or /plans/yearly/{SITE_SLUG} )
+ */
+const usePlanTypeDestinationCallback = () => {
+	return useCallback(
+		( props: Partial< PlanTypeSelectorProps >, additionalArgs: PathArgs = {} ) => {
+			const { intervalType = '' } = additionalArgs;
+			const defaultArgs = {
+				customerType: undefined,
+				discount: props.withDiscount,
+				feature: props.selectedFeature,
+				plan: props.selectedPlan,
```

Comment:
I wonder (I haven't checked, sorry) if these props are only needed for the callback here and nowhere else in the plan-type-selector component, in which case we can remove them from `PlanTypeSelectorProps` and minimize that interface a bit.

Reply:
Great question. I also noticed some oddness yesterday https://github.com/Automattic/wp-calypso/pull/86127#discussion_r1449739205. I'll do some quick auditing today before merging

Reply:
`withDiscount`, `selectedFeature`, and `selectedPlan` appear to only be used by the callback. I'll remove them from `PlanTypeSelectorProps` when I refactor the plan type selector toggle in the immediate follow-up I was planning